src/sentry/attachments/base.py

```diff
@@ -153,19 +161,45 @@ def get(self, key):
             attachment.setdefault("key", key)
             yield CachedAttachment(cache=self, **attachment)
 
-    def get_data(self, attachment):
-        data = []
+    def get_data(self, attachment) -> bytes:
+        data = bytearray()
 
         for key in attachment.chunk_keys:
             raw_data = self.inner.get(key, raw=True)
             if raw_data is None:
                 raise MissingAttachmentChunks()
-            data.append(zlib.decompress(raw_data))
+            if raw_data.startswith(b"\x28\xb5\x2f\xfd"):
+                decompressed = zstandard.decompress(raw_data)
+            else:
+                decompressed = zlib.decompress(raw_data)
+            data.extend(decompressed)
 
-        return b"".join(data)
+        return bytes(data)
+
+    def get_zstd_data(self, attachment) -> bytes:
+        data = bytearray()
+
+        for key in attachment.chunk_keys:
+            raw_data = self.inner.get(key, raw=True)
+            if raw_data is None:
+                raise MissingAttachmentChunks()
+            if raw_data.startswith(b"\x28\xb5\x2f\xfd"):
+                data.extend(raw_data)
+            else:
+                decompressed = zlib.decompress(raw_data)
+                compressed = zstandard.compress(decompressed)
+                data.extend(compressed)
+
+        return bytes(data)
 
     def delete(self, key):
         for attachment in self.get(key):
             attachment.delete()
 
         self.inner.delete(ATTACHMENT_META_KEY.format(key=key))
+
+
+def compress_chunk(chunk_data: bytes) -> bytes:
+    if options.get("attachment-cache.use-zstd"):
+        zstandard.compress(chunk_data)
```

Comment:
Shouldn't this return here?

```suggestion
        return zstandard.compress(chunk_data)
```

Reply:
lol, right :-D